libs/utils/utils.js

```diff
@@ -165,13 +163,39 @@ export function createTag(tag, attributes, html) {
   return el;
 }
 
-export function makeRelative(href) {
-  const fixedHref = href.replace(/\u2013|\u2014/g, '--');
-  const hosts = [`${PROJECT_NAME}.hlx.page`, `${PROJECT_NAME}.hlx.live`, ...PRODUCTION_DOMAINS];
-  const url = new URL(fixedHref);
-  const relative = hosts.some((host) => url.hostname.includes(host))
-    || url.hostname === window.location.hostname;
-  return relative ? `${url.pathname}${url.search}${url.hash}` : href;
+function getExtension(path) {
+  const pageName = path.split('/').pop();
+  return pageName.includes('.') ? pageName.split('.').pop() : '';
+}
+
+export function localizeLink(href, originHostName = window.location.hostname) {
+  const url = new URL(href);
+  const relative = url.hostname === originHostName;
+  const processedHref = relative ? href.replace(url.origin, '') : href;
+  const { hash } = url;
+  if (hash === '#_dnt') {
+    return processedHref.split('#')[0];
+  }
+  const path = url.pathname;
+  const extension = getExtension(path);
+  const allowedExts = ['', 'html', 'json'];
+  if (!allowedExts.includes(extension)) {
+    return processedHref;
+  }
```

Comment:
I _think_ this should work in a few places...
```suggestion
  if (!allowedExts.includes(extension) return processedHref;
```
Right, @chrischrischris?

Reply:
Done. :)